### PR TITLE
Fix usability issues with scene tabs

### PR DESCRIPTION
**Pull request by YeldhamDev**
- Stop scene creation on double-click when hovering the offset buttons.
- Fix the "Add Scene" button's position on RTL mode.

### CODE REVIEW

**Comment by KoBeWi** [comment]
> Stop scene creation on double-click when hovering the offset buttons.

Does not seem fixed, unless I'm misunderstanding something:

https://github.com/godotengine/godot/assets/2223172/8e4a6e16-3c86-4024-ad56-17223a61cde3

**Comment by YeldhamDev** [comment]
@KoBeWi Oops! Forgot to delete a line, try again.

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -5679,11 +5679,17 @@ void EditorNode::_scene_tab_input(const Ref<InputEvent> &p_input) {
 			if (mb->get_button_index() == MouseButton::MIDDLE && mb->is_pressed()) {
 				_scene_tab_closed(scene_tabs->get_hovered_tab());
 			}
-		} else {
-			if (mb->get_button_index() == MouseButton::LEFT && mb->is_double_click()) {
+		} else if (mb->get_button_index() == MouseButton::LEFT && mb->is_double_click()) {
+			int tab_buttons = 0;
+			if (scene_tabs->get_offset_buttons_visible()) {
+				tab_buttons = theme->get_icon(SNAME("increment"), SNAME("TabBar"))->get_width() + theme->get_icon(SNAME("decrement"), SNAME("TabBar"))->get_width();
```
Comment (KoBeWi) [inline editor/editor_node.cpp:5685]: I wonder if this couldn't be done better, e.g. as a TabBar method.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -5679,11 +5679,17 @@ void EditorNode::_scene_tab_input(const Ref<InputEvent> &p_input) {
...
+			}
+
+			if ((gui_base->is_layout_rtl() && mb->get_position().x > tab_buttons) || (!gui_base->is_layout_rtl() && mb->get_position().x < scene_tabs->get_size().width - tab_buttons)) {
```
Comment (KoBeWi) [inline editor/editor_node.cpp:5688]: I wonder if this couldn't be done better, e.g. as a TabBar method.

**Review by KoBeWi — APPROVED** [review]

File: editor/editor_node.cpp (hunk 2 above)
Comment (YeldhamDev) [reply editor/editor_node.cpp:5688]: Feels like a very specific use case.

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**